### PR TITLE
os/bluestore: cleanup BitAllocator

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

**Comment by xiexingguo** [comment]
@chhabaramesh Can you look?

File: src/os/bluestore/BitAllocator.cc [hunk 1]
```diff
@@ -1311,6 +1311,7 @@ BitAllocator::shutdown()
 {
   lock_excl();
   serial_lock();
+  serial_unlock();
```
Comment (varadakari) [inline src/os/bluestore/BitAllocator.cc]: what is this method doing? I am not able to get anything except lock() and unlock(). If we want to make sure no one is holding the lock then we should make sure, with a bool or counter.

**Comment by chhabaramesh** [comment]
@xiexingguo , the changes look good to me. thanks for taking this up.

**Comment by chhabaramesh** [comment]
@varadakari  @xiexingguo , this function is just to hide the mode of Allocator and unconditionally do the required locking when in serial or concurrent mode.

**Comment by chhabaramesh** [comment]
@xiexingguo please make sure: unittest_bit_alloc, unittest_alloc, ceph_test_objectstore passes with your changes :-).

**Comment by varadakari** [comment]
@chhabaramesh But that is shutdown method, i don't follow you on hiding the mode of allocator, that shouldn't matter in shutdown time right? or i am not following the shutdown correctly here.

**Comment by chhabaramesh** [comment]
@varadakari , I did not notice it was in shutdown call. The reason of taking lock here and unlock is to make sure we wait for any on going calls before doing shutdown.

**Comment by xiexingguo** [comment]
Passed all local tests:
```
[----------] 67 tests from ObjectStore/StoreTest (4829982 ms total)

[----------] Global test environment tear-down
[==========] 67 tests from 1 test case ran. (4829983 ms total)
[  PASSED  ] 67 tests.

**Comment by xiexingguo** [comment]
> we should assert or wait for all the threads to complete in a meaningful way.

Done.

**Pull request merged**